WordPress-Core/ruleset.xml

```diff
@@ -150,6 +150,9 @@
 		<!-- Covers rule: When type casting, do it like so: $foo = (boolean) $bar; -->
 		<rule ref="Generic.Formatting.SpaceAfterCast"/>
 		<rule ref="WordPress.WhiteSpace.CastStructureSpacing"/>
+		<rule ref="WordPress.WhiteSpace.CastStructureSpacing.ContainsWhiteSpace">
+			<type>warning</type>
```

Comment:
> Why does this need to be a warning instead of an error?

This PR was about syncing with the upstream sniff and was not intended to *change* existing functionality.
The WPCS version of the sniff had this as a `warning` previously, so I figured there must have been a reason at the time the sniff was copied into WPCS for this check to have been changed from `error` (upstream) to `warning` (WPCS), which is why I left it as such.

> Is it not called for by WordPress's standards?

It is: https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage

> I suggest that we let it be an error, since as I understand it it already is right now, no?

As mentioned above, currently it's a `warning`, but I'd be happy to change it to an `error`.